src/alacris_lib/side_chain_user.ml

```diff
@@ -61,24 +62,49 @@ let get_keypair_of_address user =
   Lwt_exn.catching_arr keypair_of_address user
 
 
-module ContractAddrType = struct
-  [@warning "-39"]
-  type t = { contract_address : Address.t }
-  [@@deriving yojson {strict = false}]
-  include (YojsonPersistable (struct
-             type nonrec t = t
-             let yojsoning = {to_yojson;of_yojson}
-           end) : (PersistableS with type t := t))
-end
-
 
 let get_contract_address_from_client_exn_req : unit -> Address.t Lwt_exn.t =
+  fun () ->
+  let contract_address = Lazy.force contract_address_client in
+  Lwt_exn.return contract_address
+
+
+let get_contract_address_from_client_checked_exn_req : unit -> Address.t Lwt_exn.t =
   fun () ->
   let open Lwt_exn in
-  UserQueryRequest.Get_contract_address
-    |> post_user_query_request
-  >>= fun x ->
-    return (ContractAddrType.of_yojson_exn x).contract_address
+  get_contract_address_from_client_exn_req ()
+  >>= fun contract_address ->
+  (* We are in the strange situation that the contract code as obtained from the web3.eth.getCode
+     does not match the Operator_contract_binary.contract_bytes
+     If we decrease the index by 1, then there is no contract. So we clearly download the
+     latest contract.
+     We need to understand the contruction of the code itself. Apparently the values are prefixed
```

Comment:
Typo: `contruction` should be `construction`

Reply:
Done.